### DISCUSSION TITLE
Every sensor shows correct data, but after 30 seconds every sensor change to unavailable

### DISCUSSION THREAD

**Discussion by nkk67** [wlcrs/huawei_solar#562] [Connecting to the inverter]
Connection to my inverter works perfect Modbus TCP Port 502. And after installation or reloading every sensor shows the correct value. But after about 15 to 30 seconds every sensor become "unavailable". After reload all sensors show again the correct values.
I have a SUN2000-12KTL-M5 with V200R022C10SPC104 Firmware and a SDongleA-05 with V200R022C10SPC108 FIrmware. The Developer told me that this is an unstable connection to the inverter. I tried a lot, but could not find any solution.

**Comment by esalamasip** [comment]
Hello, same problem happens to me after installation. I'm using EW11a, port 502, ID=1. The integration and the entities are loaded but all sensors  are shown as Unavailable and I can't collect any data. Any suggestion would be welcome. Thanks.

**Comment by nkk67** [comment]
I solved the problem by restricting the modbus TCP in my huawei inverter to only one ip address. There is a menu where you can enable modbus, unrestricted or restricted to one ip. When restricted to the ip of my HA then everything ist loaded and ok.

**Reply by steinaaj** [reply]
THIS!!! THANK YOU SO MUCH!!! I've had a plant, where it kept changing to unavailable a few seconds after the add in HA - this did the trick! 🥇
